orchagent/portsorch.cpp

```diff
@@ -2357,10 +2343,19 @@ void PortsOrch::initPortCapLinkTraining(Port &port)
     SWSS_LOG_WARN("Unable to get %s LT support capability", port.m_alias.c_str());
 }
 
-bool PortsOrch::isFecModeSupported(const Port &port, sai_port_fec_mode_t fec_mode)
+bool PortsOrch::isFecModeSupported(const Port &port, sai_port_fec_mode_t fec_mode, bool is_auto)
```

Comment:
@dgsudharsan for FEC override you do have a separate capability. Why not to have a separate method to check whether this mode is supported or not? It can be done while processing port configuration attributes in the main loop, while having a distinction based on `autoneg` configuration

Reply:
@nazariig  The FEC override configuration is not related to autoneg. We need to have a single place to check if a specific FEC mode is supported. Due to refactoring this method parameter has changed from string to sai_port_fec_mode_t. We need a flag to differentiate auto vs none.